### PR TITLE
fix: Close #189556, gulp task do not display task name

### LINKED ISSUE #189556: gulp task do not display task name
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version:  

版本: 1.80.1 (Universal)
提交: 74f6148eb9ea00507ec113ec51c489d6ffb4b771
日期: 2023-07-12T17:20:23.298Z
电子: 22.3.14
ElectronBuildId: 21893604
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Darwin x64 21.3.0

- OS Version: Mac os 12.2.1 

Steps to Reproduce:

1. Clone vscode source code 
2. yarn install and use `shift + command + b` to call `watch-client` and `watch-extensions` task
3. some gulp task show `anonymous`

image

### PR DESCRIPTION
**Pull request by yiliang114**
fixes #189556

After add Task Name

image

### CODE REVIEW

**Review by meganrogge — DISMISSED** [review]

**Comment by meganrogge** [comment]
> The git repo is not clean after compiling the /build/ folder. Did you forget to commit .js output for .ts files?

the above error is present

File: build/lib/compilation.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ function createCompile(src: string, build: boolean, emitError: boolean, transpil
 
 export function transpileTask(src: string, out: string, swc: boolean): () => NodeJS.ReadWriteStream {
```
Comment (alexr00) [inline build/lib/compilation.ts:96]: The return type doesn't seem correct anymore.

File: build/lib/compilation.ts [hunk 2]
```diff
@@ -104,11 +104,14 @@ export function transpileTask(src: string, out: string, swc: boolean): () => Nod
 			.pipe(transpile())
 			.pipe(gulp.dest(out));
 	};
+
+	task.taskName = `transpile-${path.basename(src)}`;
+	return task;
 }
 
 export function compileTask(src: string, out: string, build: boolean, options: { disableMangle?: boolean } = {}): () => NodeJS.ReadWriteStream {
```
Comment (alexr00) [inline build/lib/compilation.ts:112]: Same for this. It looks like the return type should be 
```
{
    (): NodeJS.ReadWriteStream;
    taskName: string;
}
```
 for both of these methods.

**Review by alexr00 — COMMENTED** [review]
The return type of the changed functions seems fishy to me.

File: build/lib/compilation.ts (hunk 2 above)
Comment (yiliang114) [reply build/lib/compilation.ts]: > Same for this. It looks like the return type should be
> 
> ```
> {
>     (): NodeJS.ReadWriteStream;
>     taskName: string;
> }
> ```
> 
> for both of these methods.

Using this existing type `StreamTask` from  https://github.com/microsoft/vscode/blob/85daa35afb86202badbee79532537bf226961f25/build/lib/task.ts#L17 seems more appropriate?

**Review by alexr00 — DISMISSED** [review]
Looks good to me, though @meganrogge I don't think this has much to do with VS Code tasks. As far as I can tell the printing of the name is here: 
https://github.com/yiliang114/vscode/blob/818ae5029a3792e31804be7ad83e30c42f063eb0/build/lib/task.ts#L38-L39

**Comment by meganrogge** [comment]
it's still failing. @alexr00 looks like @alexdima added this code originally, so maybe he should have a look, though it's been awhile

**Comment by meganrogge** [comment]
```
The git repo is not clean after compiling the /build/ folder. Did you forget to commit .js output for .ts files?
```

**Comment by yiliang114** [comment]
> ```
> The git repo is not clean after compiling the /build/ folder. Did you forget to commit .js output for .ts files?
> ```

Sorry, Let me confirm. You mean this file should not be submitted, right? Because it is compiled.  

image

I think the previous submissions seem to have submitted the compiled products. However, it's okay. If the repo is kept clean, it really cannot be submitted.

**Comment by yiliang114** [comment]
> maybe try removing this file from the changes?
> 
> `build/lib/compilation.js`

Okay, I 've already done that

**Review by meganrogge — APPROVED** [review]

**Comment by meganrogge** [comment]
Thanks for your work @yiliang114 !

**Review by connor4312 — APPROVED** [review]

**Pull request merged**